kernel/include/ksched.h

```diff
@@ -13,289 +13,108 @@
-extern int _is_thread_time_slicing(struct k_thread *thread);
-extern void _update_time_slice_before_swap(void);
-#ifdef _NON_OPTIMIZED_TICKS_PER_SEC
-extern s32_t _ms_to_ticks(s32_t ms);
-#endif
-extern void idle(void *, void *, void *);
-
-/* find which one is the next thread to run */
-/* must be called with interrupts locked */
-#ifdef CONFIG_SMP
-extern struct k_thread *_get_next_ready_thread(void);
-#else
-static ALWAYS_INLINE struct k_thread *_get_next_ready_thread(void)
-{
-	return _ready_q.cache;
-}
-#endif
-
-static inline int _is_idle_thread(void *entry_point)
-{
-	return entry_point == idle;
-}
-
-static inline int _is_idle_thread_ptr(k_tid_t thread)
-{
-#ifdef CONFIG_SMP
-	return thread->base.is_idle;
-#else
-	return thread == _idle_thread;
-#endif
-}
-
-/*
- * The _is_prio_higher family: I created this because higher priorities are
- * lower numerically and I always found somewhat confusing seeing, e.g.:
- *
- *   if (t1.prio < t2.prio) /# is t1's priority higher then t2's priority ? #/
- *
- * in code. And the fact that most of the time that kind of code has this
- * exact comment warrants a function where it is embedded in the name.
- *
- * IMHO, feel free to remove them and do the comparison directly if this feels
- * like overkill.
- */
-
-static inline int _is_prio1_higher_than_or_equal_to_prio2(int prio1, int prio2)
-{
-	return prio1 <= prio2;
-}
-
-static inline int _is_prio_higher_or_equal(int prio1, int prio2)
-{
-	return _is_prio1_higher_than_or_equal_to_prio2(prio1, prio2);
-}
-
-static inline int _is_prio1_higher_than_prio2(int prio1, int prio2)
-{
-	return prio1 < prio2;
-}
-
-static inline int _is_prio_higher(int prio, int test_prio)
-{
-	return _is_prio1_higher_than_prio2(prio, test_prio);
-}
-
-static inline int _is_prio1_lower_than_or_equal_to_prio2(int prio1, int prio2)
-{
-	return prio1 >= prio2;
-}
-
-static inline int _is_prio_lower_or_equal(int prio1, int prio2)
-{
-	return _is_prio1_lower_than_or_equal_to_prio2(prio1, prio2);
-}
-
-static inline int _is_prio1_lower_than_prio2(int prio1, int prio2)
-{
-	return prio1 > prio2;
-}
-
-static inline int _is_prio_lower(int prio1, int prio2)
-{
-	return _is_prio1_lower_than_prio2(prio1, prio2);
-}
-
-static inline int _is_t1_higher_prio_than_t2(struct k_thread *t1,
-					     struct k_thread *t2)
-{
-	return _is_prio1_higher_than_prio2(t1->base.prio, t2->base.prio);
-}
-
-static inline int _is_higher_prio_than_current(struct k_thread *thread)
-{
-	return _is_t1_higher_prio_than_t2(thread, _current);
-}
-
 #ifdef CONFIG_MULTITHREADING
-static inline int _is_valid_prio(int prio, void *entry_point)
-{
-	if (prio == K_IDLE_PRIO && _is_idle_thread(entry_point)) {
-		return 1;
-	}
-
-	if (!_is_prio_higher_or_equal(prio,
-				      K_LOWEST_APPLICATION_THREAD_PRIO)) {
-		return 0;
-	}
-
-	if (!_is_prio_lower_or_equal(prio,
-				     K_HIGHEST_APPLICATION_THREAD_PRIO)) {
-		return 0;
-	}
-
-	return 1;
-}
+#define _VALID_PRIO(prio, entry_point) \
```

Comment:
Heh. I just changed this the other day (by writing `_is_valid_prio()`) to fix some Coverity whining.